src/ape/utils/process.py

```diff
@@ -0,0 +1,49 @@
+import queue
+import threading
+import time
+
+from ape.exceptions import SubprocessTimeoutError
+
+
+class JoinableQueue(queue.Queue):
+    """
+    A queue that can be joined, useful for multi-processing.
+    Borrowed from the ``py-geth`` library.
+    """
+
+    def __iter__(self):
+        while True:
```

Comment:
Need to think of edge cases that could cause infinite loop here, not saying there is, but just in case. This is just a preliminary review so far, taking note